store/src/test/java/org/apache/rocketmq/store/DefaultMessageStoreTest.java

```diff
@@ -657,6 +658,30 @@ public void testCleanUnusedLmqTopic() throws Exception {
 
     }
 
+    @Test
+    public void testEncodeLongMessage() throws Exception{
+        MessageExtBrokerInner messageExtBrokerInner = buildMessage();
+        Class clazz = messageStore.getClass();
+
+        Field fieldCommitLog = clazz.getDeclaredField("commitLog");
+        fieldCommitLog.setAccessible(true);
+        CommitLog commitLog = (CommitLog) fieldCommitLog.get(messageStore);
+
+        Field fieldMessageStoreConfig = clazz.getDeclaredField("messageStoreConfig");
+        fieldMessageStoreConfig.setAccessible(true);
+        MessageStoreConfig messageStoreConfig = (MessageStoreConfig) fieldMessageStoreConfig.get(messageStore);
+        messageExtBrokerInner.setBody(new byte[messageStoreConfig.getMaxMessageSize()]);
+
+        CommitLog.PutMessageThreadLocal putMessageThreadLocal = commitLog.getPutMessageThreadLocal().get();
+        PutMessageResult encodeResult1 = putMessageThreadLocal.getEncoder().encode(messageExtBrokerInner);
+        assertTrue(encodeResult1 == null);
+
```

Comment:
@shengminw This test is for DefaultMessageStore.
So just use type conversion (DefaultMessageStore) messagestore, instead of reflection.

The reflection is not friendly for further evolution.

Reply:
BTW, it maybe need another test for body size exactly equal to max message size.

Reply:
ok, I will revise it.